### DISCUSSION TITLE
.scss.module in bundle (using analyze)

### DISCUSSION THREAD

**Discussion by joaogarin** [vercel/next.js#44681] [Help]
Hello,

I am using css modules with sass and noticing a lot of scss modules inside the bundle when I run `yarn analyze`(notice also in build).

![Screenshot from 2023-01-07 11-52-01](https://user-images.githubusercontent.com/2866604/211153282-d62fc85c-6f6d-4c25-bdcb-77ae7ccc142d.png)

This is all inside _app.js bundle.

Some of these are up to 8KB and 14KB gziped which is HUGE. Any ideas how this can even be? at the moment I am not purging unused css and this seems to be related to it, however even without purging how can this affect the bundle size so much?

**Comment by GabenGar** [comment]
You aren't using [sass module  imports](https://sass-lang.com/documentation/at-rules/use), therefore every single file is built as a separate entrypoint, with contents of all their `@import`s  included in every single file.

**Reply by joaogarin** [reply]
Do you mean doing `@use file.scss` instead of `@import file.scss` inside the .scss files?

**Reply by GabenGar** [reply]
Yes, but not only that. You have to ensure the modules imported by `@use` also do not include actual css rules, as it will be included in the final css of the module. As far as sass modules concerned you are building `n` separate entry points, where `n` is the amount of css modules, not the single module tree.

**Reply by joaogarin** [reply]
We tried several approaches around this, but it didn' really solve the problem. The main puzzling thing is why this dead code is in js files and not in css files.. I guess that is some internal thing to css modules itself. but what we are ending up with is huge js files ,not css files (Whcih I would unddestand more if that was the case)

seems rleated to this topic https://github.com/css-modules/css-modules/issues/310

**Reply by GabenGar** [reply]
At this point you should try to create a reproduction repo. The css module mappings could only be an issue if you store all declarations in a single file which gets imported into all other modules.

**Reply by joaogarin** [reply]
Thanks for the response and feedback here, I appreciate it!

I have created a reproduction here https://github.com/joaogarin/css-modules-bundle/tree/main basically I see this behavior happening even without any sass `@import` (hello world class is present in the bundle, and the more classes / styles you have this grows larger and larger) https://github.com/css-modules/css-modules/issues/310

in my app I already tried swapping all imports with use statements I see no changes as well..The bundle is massive, and its pretty much all .module.scss files.

I see other mentions on this but I think its just not clear as to maybe why this mapping is necessary. Perhaps it is but would be good to know since people usually are choosing css modules for the bundle concerns. I dont think this maps to bad performance per-se but still seeing 300KB of styles in the bundle is not great :/
